src/modules/txbuilder/RadixTransactionBuilder.ts

```diff
@@ -372,12 +384,17 @@ export default class RadixTransactionBuilder {
         name: string,
         symbol: string,
         description: string,
-        granularity: BN,
+        granularity: number | string | Decimal,
         decimalQuantity: number | string | Decimal,
         iconUrl: string,
         permissions: RadixTokenPermissions,
     ) {
         const tokenAmount = this.getSubUnitsQuantity(decimalQuantity)
+        const tokenGranularity = this.getSubUnitsQuantity(granularity)
+
+        if (tokenAmount.lt(this.BNZERO)) {
+            throw new Error('Negative quantity is not allowed')
+        }
```

Comment:
I see we are validating the case where `subunits == zero` as well in above code (not only `< zero`). Would it make sense to validate the `tokenAmount == zero` case too here? Or is this enforced somewhere else later in the chain?

Reply:
This is in the `createToken` method - it should be perfectly fine to create a token with 0 initial supply that you plan to mint later(I actually encountered this issue in my Radflix project). The only consideration is a single issuance token, not sure if we want to allow that or not, currently there's no use case for it, but if they become transferrable, then it could be used as an NFT effectively, not sure

Reply:
Sounds like a no-issue to me then.

Reply:
Added a 0-amount check for single issuance tokens